src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -447,20 +447,26 @@ class UsersServiceImpl : KoinComponent, UsersService {
         }
 
         val identityGroup = gson.fromJson(org.metadata.data(), IdentityGroup::class.java)
-        val encodedEmail = if (subOrganizationId != null) {
-            encodeSubOrgUserEmail(
-                email,
-                organizationId,
-                subOrganizationId
-            )
-        } else {
-            email
-        }
+        val encodedEmail = getEmail(organizationId, subOrganizationName, email)
         identityProvider.authenticate(identityGroup, encodedEmail, password)
 
         return getUser(ResourceHrn(userRecord.hrn), userRecord)
     }
 
+    private fun getEmail(
+        organizationId: String,
+        subOrganizationName: String?,
+        email: String,
+    ) = if (subOrganizationName != null) {
+        encodeSubOrgUserEmail(
+            email,
+            organizationId,
+            subOrganizationName
+        )
+    } else {
+        email
+    }
```

Comment:
```suggestion
    ) = subOrganizationName?.let {
      encodeSubOrgUserEmail(it, organizationId, subOrganizationName)
    } ?: email

```

Reply:
Addressed in this PR - https://github.com/hwslabs/iam/pull/163